vision/module/in_frame.py

```diff
@@ -43,16 +46,23 @@ def ModuleInFrame(img):
     laplacian = np.uint8(laplacian)
 
     # Hough Circle Detection
-    # circles = (x, y, r)
-    circles = cv2.HoughCircles(image=laplacian, method=cv2.HOUGH_GRADIENT, dp=1, minDist=8, param1=50, param2=40, minRadius=0, maxRadius=50)
+    circles = cv2.HoughCircles(image=laplacian, method=cv2.HOUGH_GRADIENT, dp=1, minDist=8, param1=50, param2=28, minRadius=0, maxRadius=50)
+    if circles is None:  # no circles found
+        return False
+    elif circles.shape[1] > 100:  # too many circles found
+        raise ValueError("Too many circles found (" + str(circles.shape[1]) + ")")
+
     circles = np.uint16(circles)
 
     # Resize circles into 2d array
     circles = np.reshape(circles, (np.shape(circles)[1], 3))
 
     # Finding slopes between the circles
     slopes = np.array([])
-    for x, y, _ in circles:
+
+    for x, y, r in circles:
+        cv2.circle(output, (x, y), r, (0, 255, 0), 4)
+        cv2.rectangle(output, (x - 5, y - 5), (x + 5, y + 5), (0, 128, 255), -1)
```

Comment:
Lets not modify the image unless in debugging mode because these will slow down the algorithm.